LoopKit/SettingsStore.swift

```diff
@@ -276,6 +285,7 @@ public struct StoredSettings: Equatable {
                 cgmDevice: HKDevice? = nil,
                 pumpDevice: HKDevice? = nil,
                 bloodGlucoseUnit: HKUnit? = nil,
+                dosingStrategy: DosingStrategy = .tempBasalOnly,
```

Comment:
Hm... I see now that this may be intentionally limited to the strategy for automatic dosing. If this is correct, the name is very confusing at it is. With a name change, the case names could also be updated.

Reply:
We don't have anything that represents a "strategy" for manual dosing (and I don't think that would make sense; as that's in the user's head), so I probably lean towards keeping the name shorter, but if you feel strongly that changing the name to "AutomaticDosingStrategy" is more clear, I can do that.

Reply:
Yes, a more specific name help avoid confusion. Possible temp basal or bolus recommendations are part of the dosing strategy. 

Do you agree that `AutomaticDosingStrategy` makes it clearer?

Reply:
I thought DosingStrategy was specific enough (as "manual" dosing strategy isn't a thing we need to distinguish from), but I'm obviously not "fresh eyes", so I can change it as it sounds like you find it more clear, and I do think the longer name is still accurate.

Reply:
Also, this setting and these possible values have been around in the code a long time; this is just adding it to SettingStore.

Reply:
Yeah, I'm not being facetious in saying I think that `DosingStrategy` is a clear name, but I named it, and so it's a product of my own headspace, and thus I'm not necessarily a good judge of how clear it will be for newcomers. I'll dig into updating this name.

As for the names of the strategies themselves, I don't think it's the job of the name to carry all the details about how the strategy works, but just that it's clear enough to identify the strategy, so I'd prefer to keep those the same. I don't want it to say "automaticBolusOfFortyPercentNeededDoseForInsulinAdditionWithLowTempBasalsForInsulinWitholding Strategy" and "automaticTempBasalOf100PercentNeededDoseOverThirytMinutesWithResetAfterTwentyMinutesAndNoAutomaticBolus Strategy".  Those names *are* facetious, of course, but I'm just trying to illustrate that the actual names are more about distinguishing identity.  The automatic bolus strategy *does* still use low temp basals for reducing insulin delivery.

Reply:
> Possible I didn't see it in previous PRs. Sorry about that if I did over look it.

This was probably in Loop before you were at Tidepool, so no worries. You also probably didn't need to look at any of that code as Tidepool Loop only uses the original temp basal strategy, and not Automatic Bolus, and "Automatic Bolus" is also a name that is widely known in the DIY community, but not much outside it.

Reply:
"Automatic Bolus" is the name of the strategy, and has been in common usage in the DIY community for years now. Nobody calls it "bolus with temp basal".  The context that this name came about is that all boluses were manual before it existed. And so when it was introduced, the name "Automatic Bolus" made a lot of sense, and is now *the* name of this strategy. Renaming it would require documentation changes, and cause a lot of confusion in the community, I think. If we just changed the swift case name, then we'd have an odd mismatch between the way it was referred to in code vs how it was referred to in common use.